tests/test_integrated_channels/test_api/test_degreed/test_views.py

```diff
@@ -0,0 +1,106 @@
+        self.assertEqual(data.get('degreed_user_id'), self.degreed_config.degreed_user_id)
+        self.assertEqual(data.get('degreed_user_password'),
+            self.degreed_config.degreed_user_password)
+        self.assertEqual(data.get('enterprise_customer'),
+                         str(self.degreed_config.enterprise_customer.uuid))
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_update(self, mock_current_request):
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:degreed:configuration-detail', args=[self.degreed_config.id])
+        payload = {
+            'degreed_base_url': 'http://testing2',
+            'degreed_company_id': 'test',
+            'enterprise_customer': ENTERPRISE_ID,
+            'degreed_user_id': 893489,
+            'degreed_user_password': 'password',
+            'key': 'testing',
+            'secret': 'secret',
+        }
+        response = self.client.put(url, payload)
+        self.degreed_config.refresh_from_db()
+        self.assertEqual(self.degreed_config.degreed_base_url, 'http://testing2')
+        self.assertEqual(self.degreed_config.degreed_company_id, 'test')
+        self.assertEqual(self.degreed_config.key, 'testing')
+        self.assertEqual(self.degreed_config.secret, 'secret')
+        self.assertEqual(self.degreed_config.degreed_user_password, 'password')
+        self.assertEqual(self.degreed_config.degreed_user_id, '893489')
+        self.assertEqual(response.status_code, 200)
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_patch(self, mock_current_request):
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:degreed:configuration-detail', args=[self.degreed_config.id])
+        payload = {
+            'degreed_base_url': 'http://testingchange',
+            'enterprise_customer': ENTERPRISE_ID,
+        }
+        response = self.client.patch(url, payload)
+        self.degreed_config.refresh_from_db()
+        self.assertEqual(self.degreed_config.degreed_base_url, 'http://testingchange')
+        self.assertEqual(response.status_code, 200)
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_delete(self, mock_current_request):
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:degreed:configuration-detail', args=[self.degreed_config.id])
+        response = self.client.delete(url)
+        configs = DegreedEnterpriseCustomerConfiguration.objects.filter()
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(len(configs), 0)
```

Comment:
ideally should there be a check for this length before the delete is issue to be further strict?

Reply:
? I don't think I quite follow. The url passed to the delete function includes an id. If we had 0 configs to start with, then that id would be invalid and throw an error.

Reply:
I think he means check the length of `configs` before you do the delete as you do afterwards. We could also add another so that we have two and go to one, either way.

Reply:
ah maybe I am mixed up then, I assumed there would be a len(configs) of 1 before doing the delete request

Reply:
but given that you had indicated if the record does not exist it wil fail anyway I am not too worried just that if it's easy that kind of test is a bit more explicit.. but maybe not worth for this small bit

Reply:
Gotcha. Yeah specifying 1 is definitely more explicit, but yeah like I said we wouldn't even have an id to delete and it'd throw an error.